src/sentry/static/sentry/app/views/usageStats/usageChart/index.tsx

```diff
@@ -211,11 +227,25 @@ export class UsageChart extends React.Component<Props, State> {
 
     const {label, value} = selectDataCategory;
 
+    // If date intervals is in hours, a wider x-axis interval is needed for the
+    // labels so that dates are not truncated
+    //
+    // For hourly: we want to put 5 bars between each tick and 23 bars between
+    // each label, so 1 tick == 6 hours and 1 label == 24 hours
+    //
+    // For weekly: we want to put 6 bars between each tick and 6 bars between
+    // each label, so 1 tick == 7 days and 1 label == 7 days
+    const isHourly = usageDateInterval.replace(/[0-9]/g, '') === 'h';
```

Comment:
Will it be possible that `usageDateInterval` could be defined as a large number of hours or just 1 day? Or (e.g `1d` or `3000h`). There are some utils in `sentry/utils/dates` that parse this string into the actual time duration.